src/pytorch_lightning/utilities/data.py

```diff
@@ -321,27 +327,39 @@ def _dataloader_init_kwargs_resolve_sampler(
     fault_tolerant_mode = _FaultTolerantMode.detect_current_mode()
     batch_sampler = getattr(dataloader, "batch_sampler")
     is_predicting = mode == RunningStage.PREDICTING
-    # checking the batch sampler type is different than PyTorch default.
-    if batch_sampler is not None and (type(batch_sampler) is not BatchSampler or is_predicting):
-        batch_sampler = type(batch_sampler)(
-            sampler,
-            batch_size=batch_sampler.batch_size,
-            drop_last=(False if is_predicting else batch_sampler.drop_last),
-        )
-        if is_predicting:
-            batch_sampler = IndexBatchSamplerWrapper(batch_sampler)
-
-        if fault_tolerant_mode.is_automatic:
-            fast_forward_sampler = batch_sampler = FastForwardSampler(batch_sampler)
-            fast_forward_sampler.setup(dataloader_batch_size=1)
-
-        return {
-            "sampler": None,
-            "shuffle": False,
-            "batch_sampler": batch_sampler,
-            "batch_size": 1,
-            "drop_last": False,
-        }
+
+    if batch_sampler is not None:
+        if disallow_batch_sampler:
+            # Check that we don't have a PyTorch default batch sampler that was instantiated in DataLoader __init__
+            if not (
+                type(batch_sampler) is BatchSampler
+                and batch_sampler.sampler == sampler
+                and dataloader.batch_size == batch_sampler.batch_size
+            ):
+                raise MisconfigurationException(
+                    "It is not possible to have a batch sampler in your dataloader, "
+                    "when running on multiple IPU devices."
+                )
```

Comment:
@otaj I think something got missed here. There is no check anywhere that we are dealing with IPU here. 
See also the IPUStrategy where we convert to the poptorch Dataloader wrapper.

Reply:
It's a bit hidden, but the argument `disallow_batch_sampler` is by default `False` and only place where we are passing any value there is in https://github.com/Lightning-AI/lightning/pull/13854/files#diff-1de8258ec09ab5d1d13840e70c64b67e9dfc098de0eba1e9f6020ed191381cc8R234

Reply:
Okay thanks. It makes sense. 
I was blindly looking for IPU references and couldn't spot this one.